### DISCUSSION TITLE
how to troubleshoot failing direct donnections ?

### DISCUSSION THREAD

**Discussion by Zixim** [rustdesk/rustdesk#1345] [Q&A]
I have a bunch of remote hosts, on networks that I manage, that will never accept a direct connection. Always relayed.
The basic question is, how do I begin troubleshooting this ? Verbose logging ?

Some extra info : 

- I can also reach the hosts across VPNs, using their ip address. This results in a direct connection, as expected.
- Reaching them across the WAN always results in relayed connection.
- These are all behind Sonicwall routers.
- A couple of other hosts, behind consumer-grade isp supplied routers allow direct connection.
- I've tried with self-hosted & public relay servers, no visible difference.
- Posted on discord, was told that detailed logging is not a thing (yet), making troubleshooting this really hard.

Any info much appreciated!

**Comment by Cabus1982** [comment]
The same happens to me, but sometimes the connection is direct (15ms), but most of them are delayed (350ms)
